lib/util/transformer/length-based-splitter.js

```diff
@@ -66,6 +70,19 @@ class LengthBasedSplitter extends Stream.Transform {
     this._frameBufferIndex = 0;
   }
 
+  _pushBack (start, end, data) {
+    if (start === end) {
+      return;
+    }
+    const leftover = Buffer.allocUnsafe(end - start);
+    data.copy(leftover, 0, start, end);
+    process.nextTick(() => this.readableStream.unshift(leftover));
+  }
+
+  shutdown () {
+    this.isShutdown = true;
```

Comment:
it is recommended to set default property values in the constructor to improve node performance